ocaml/xapi/localdb.mli

```diff
@@ -18,12 +18,24 @@
 (** Thrown when a particular named key could not be found. *)
 exception Missing_key of string
 
-val get : string -> string
+val get : string -> string option
 (** Retrieves a value *)
 
-val get_with_default : string -> string -> string
-(** [get_with_default key default] returns the value associated with [key],
-    	or [default] if the key is missing. *)
+val get_exn : string -> string
+(** Retrieves the value for the key, raises Missing_key when the key is not
+    present *)
+
+val get_bool : string -> bool option
+(** Retrieves the value for the key, returns a value when it's found and is a
+    valid boolean, otherwise is [None] *)
+
+val get_int : string -> int option
+(** Retrieves the value for the key, returns a value when it's found and is a
+    valid int, otherwise is [None] *)
+
+val get_of_string : (string -> 'a option) -> string -> 'a option
```

Comment:
This reads a bit complicated and difficult to grasp. I believe this could be phrased as a monad .